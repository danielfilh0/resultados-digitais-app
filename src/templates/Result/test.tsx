import { screen } from '@testing-library/react'

import { Result } from '.'
import { renderWithTheme } from '@/utils/tests/helpers'

describe('<Result />', () => {
  it('should render the card download button', () => {
    const { container } = renderWithTheme(
      <Result
        name="Example Name"
        email="emailexample@example.com"
        phone="(99) 9 9999-9999"
      />
    )

    expect(
      screen.getByRole('button', { name: /baixar cartão/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should show error if the card data are inconsistent', () => {
    renderWithTheme(<Result name="" email="wrongemail" phone="12345" />)

    expect(screen.getByText(/erro/i)).toBeInTheDocument()
    expect(
      screen.queryByRole('button', { name: /baixar cartão/i })
    ).not.toBeInTheDocument()
  })
})
