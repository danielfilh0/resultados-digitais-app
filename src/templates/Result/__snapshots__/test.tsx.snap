// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> should render the card download button 1`] = `
.c2 {
  max-width: 128rem;
  width: 100%;
  margin: 0 auto;
  padding: 0 2rem;
}

.c8 {
  width: 100%;
  height: 5.2rem;
  display: flex;
  gap: 0.8rem;
  justify-content: center;
  align-items: center;
  border: 1px solid;
  background-color: #F2BF4E;
  line-height: 24px;
  font-size: 1.6rem;
  font-weight: 800;
  text-transform: uppercase;
  color: #100C27;
  letter-spacing: 0.2px;
  cursor: pointer;
  -webkit-box-shadow: 0px 6px 1px -2px;
  -moz-box-shadow: 0px 6px 1px -2px;
  box-shadow: 0px 6px 1px -2px;
  transition: 0.1s;
  flex-direction: row-reverse;
}

.c8:hover {
  background-color: #F8DCA0;
  border-color: transparent;
}

.c8[disabled] {
  background-color: #E1E4E8;
  color: #97A1AC;
  pointer-events: none;
  border-color: transparent;
  -webkit-box-shadow: 0px 6px 1px -2px #97A1AC;
  -moz-box-shadow: 0px 6px 1px -2px #97A1AC;
  box-shadow: 0px 6px 1px -2px #97A1AC;
}

.c0 {
  margin-top: 3.6rem;
  height: 100%;
  display: flex;
}

.c0 .c1 {
  display: flex;
  justify-content: center;
  gap: 9.5rem;
}

.c0 .c1 >img {
  display: none;
}

.c3 {
  width: 100%;
  max-width: 510px;
  display: flex;
  flex-direction: column;
  gap: 1.6rem;
}

.c3 a {
  color: #FFFFFF;
  display: flex;
  align-items: center;
  gap: 0.5rem;
}

.c3 a:first-of-type {
  width: max-content;
  font-size: 1.4rem;
}

.c3 a:last-of-type {
  justify-content: center;
  font-size: 1.4rem;
  font-weight: 800;
  text-transform: uppercase;
  text-align: center;
}

.c3 a:last-of-type span {
  max-width: 209px;
}

.c4 {
  width: 100%;
  height: 159px;
  padding: 1.5rem 3.6rem;
  display: flex;
  align-items: center;
  gap: 2rem;
  background-color: #FFFFFF;
  box-shadow: 0px 8px 18px 0px #21242929;
  border-radius: 2.4rem;
}

.c4 ul {
  padding: 0;
  list-style: none;
  display: flex;
  flex-direction: column;
  gap: 1.6rem;
}

.c4 img {
  width: 4.5rem;
}

.c5 {
  width: 3px;
  height: 100%;
  background-color: #D8D9FD;
}

.c6 {
  color: #000000;
  line-height: 1.7;
  font-weight: 400;
  font-size: 1.4rem;
}

.c9 {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  transform: rotate(90deg);
}

.c10 {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  transform: rotate(0deg);
}

@media (min-width: 1170px) {
  .c0 {
    margin-top: 0;
    align-items: center;
  }
}

@media (min-width: 1170px) {
  .c0 .c1 {
    align-items: center;
  }
}

@media (min-width: 1170px) {
  .c0 .c1 >img {
    display: block;
  }
}

@media (max-width: 1170px) {
  .c0 .c7 {
    height: 3.6rem;
    font-size: 1.4rem;
  }
}

@media (min-width: 1170px) {
  .c3 a:first-of-type {
    font-size: 2rem;
  }
}

@media (min-width: 1170px) {
  .c3 a:last-of-type {
    max-width: initial;
    font-size: 1.6rem;
    line-height: 5rem;
  }

  .c3 a:last-of-type span {
    max-width: unset;
  }
}

@media (min-width: 1170px) {
  .c4 {
    padding: 3.2rem 7.2rem;
    height: 247px;
  }
}

@media (min-width: 1170px) {
  .c4 ul {
    gap: 2.4rem;
  }
}

@media (min-width: 1170px) {
  .c4 img {
    width: 7.5rem;
  }
}

@media (min-width: 768px) {
  .c6 {
    font-size: 2rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1 c2"
  >
    <img
      alt="Imagem cartoom de um homem criando um quadro de pintura, fazendo uma analogia à criação de um cartão de visita"
      data-nimg="1"
      decoding="async"
      height="335"
      loading="lazy"
      src="/_next/image?url=%2Fillustration.png&w=1080&q=75"
      srcset="/_next/image?url=%2Fillustration.png&w=640&q=75 1x, /_next/image?url=%2Fillustration.png&w=1080&q=75 2x"
      style="color: transparent;"
      width="455"
    />
    <div
      class="c3"
    >
      <a
        href="/"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.66,202.34a8,8,0,0,1-11.32,11.32l-80-80a8,8,0,0,1,0-11.32l80-80a8,8,0,0,1,11.32,11.32L91.31,128Z"
          />
        </svg>
        Gerar outro cartão
      </a>
      <div
        class="c4"
      >
        <img
          alt="Logo secundária da Resultados Digitais"
          data-nimg="1"
          decoding="async"
          height="60"
          loading="lazy"
          src="/_next/image?url=%2Fshort-logo.png&w=256&q=75"
          srcset="/_next/image?url=%2Fshort-logo.png&w=96&q=75 1x, /_next/image?url=%2Fshort-logo.png&w=256&q=75 2x"
          style="color: transparent;"
          width="70"
        />
        <div
          class="c5"
        />
        <ul>
          <li
            class="c6"
          >
            Example Name
          </li>
          <li
            class="c6"
          >
            emailexample@example.com
          </li>
          <li
            class="c6"
          >
            (99) 9 9999-9999
          </li>
        </ul>
      </div>
      <button
        class="c7 c8"
        disabled=""
      >
        <span>
          Baixar Cartão
        </span>
        <div
          class="c9"
        >
          <svg
            aria-label="Arrow icon"
            fill="none"
            role="img"
            viewBox="0 0 25 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m16.385 8.143 3.974 4.013a.49.49 0 0 1 0 .688l-3.974 4.013a.48.48 0 0 1-.822-.344v-2.878H4.982a.484.484 0 0 1-.482-.487v-1.296c0-.269.216-.487.482-.487h10.581V8.487a.48.48 0 0 1 .822-.344Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </button>
      <a
        href="https://app.rdstation.com.br/signup"
        target="_blank"
      >
        <span>
          Fazer um teste grátis no RD Station Marketing
        </span>
        <div
          class="c10"
        >
          <svg
            aria-label="Arrow icon"
            fill="none"
            role="img"
            viewBox="0 0 25 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m16.385 8.143 3.974 4.013a.49.49 0 0 1 0 .688l-3.974 4.013a.48.48 0 0 1-.822-.344v-2.878H4.982a.484.484 0 0 1-.482-.487v-1.296c0-.269.216-.487.482-.487h10.581V8.487a.48.48 0 0 1 .822-.344Z"
              fill="currentColor"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
</section>
`;
